django/db/models/aggregates.py

```diff
@@ -228,3 +233,49 @@ def __init__(self, expression, sample=False, **extra):
 
     def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "VAR_SAMP"}
+
+
+class JSONArrayAgg(Aggregate):
+    function = "JSON_ARRAYAGG"
+    output_field = JSONField()
+    arity = 1
+
+    def as_sql(self, compiler, connection, **extra_context):
```

Comment:
```suggestion
    def as_sql(self, compiler, connection, **extra_context):
            if not connection.features.supports_json_field:
                raise NotSupportedError(
                    "JSONFields are not supported on this database backend."
                )
```
Been reviewing https://github.com/django/django/pull/18541/ which has a top level check for Json field support - I think this might want the same?

Reply:
yeah I think it makes sense to add a similar check here.